### PR TITLE
timeline zoom stack

### PR DESCRIPTION
**Pull request by fredyshox**
Push/pop timeline range actions. Save previous zooms when pushing new timeline range. Go back to route bounds, when no previous zoom is available.

### CODE REVIEW

File: src/components/DriveView/index.jsx [hunk 1]
```diff
@@ -52,17 +52,29 @@ class DriveView extends Component {
     this.setState({ windowWidth });
   }
 
+  onBack(zoom, currentRoute) {
+    if (zoom.previousZoom) {
```
Comment (incognitojam) [inline src/components/DriveView/index.jsx]:
```suggestion
    if (zoom.previous) {
```

**Comment by incognitojam** [comment]
It would be nice to have a couple of browser tests for this, for example:
- opening the demo route from URL:
  - assert that the back button is disabled
  - assert that clicking the close button takes you to the drive list
- opening a route URL with a shorter time range selected:
  - assert that clicking the back button takes you to the full route

File: src/components/DriveView/index.jsx (hunk 1 above)
Comment (fredyshox) [reply src/components/DriveView/index.jsx]: good catch!

**Comment by fredyshox** [comment]
I'm going to see what I can do. Not entirely familiar with this, but by browser tests you mean something like these puppeteer test cases (https://github.com/commaai/connect/tree/master/src/__puppeteer__)?

**Comment by incognitojam** [comment]
Yep that's the right place

**Comment by fredyshox** [comment]
Implemented your suggested puppeteer test cases in drive.test.js. Checking for url changes after button click should be enough for detecting expected navigation

**Pull request merged**